packages/libs/src/services/solana/SolanaWeb3Manager.ts

```diff
@@ -570,6 +583,182 @@ export class SolanaWeb3Manager {
     })
   }
 
+  /**
+   * Purchases USDC gated content
+   * @param params.id the id of the content, eg. the track ID
+   * @param params.type the type of the content, eg. "track"
+   * @param params.blocknumber the blocknumber the content was last updated
+   * @param params.splits map of address to USDC amount, used to split the price amoung several stakeholders
+   * @param params.extraAmount Extra amount in USDC wei to be distributed to the stakeholders
+   * @param params.purchaserUserId Id of the user that is purchasing the track
+   * @param params.senderAccount Should either be root solana account or user bank.
+   * @returns the transaction signature and/or an error
+   */
+  async getPurchaseContentWithPaymentRouterInstructions({
+    id,
+    type,
+    blocknumber,
+    // extraAmount = 0, TODO:
+    splits,
+    purchaserUserId,
+    senderAccount
+  }: {
+    id: number
+    type: 'track'
+    splits: Record<string, number | BN>
+    extraAmount?: number | BN
+    blocknumber: number
+    purchaserUserId?: number
+    senderAccount: PublicKey
+  }) {
+    if (!this.web3Manager) {
+      throw new Error(
+        'A web3Manager is required for this solanaWeb3Manager method'
+      )
+    }
+    if (Object.values(splits).length !== 1) {
+      throw new Error(
+        'Purchasing content only supports a single split. Specifying more splits coming soon!'
+      )
+    }
+
+    const recipientAmounts: Record<string, bigint> = Object.entries(
```

Comment:
should split allow both number and BN? seems error prone (comment for future, but I know marcus basically rewrote all this)

Reply:
+1 for consolidation.